ql/src/java/org/apache/hadoop/hive/ql/io/orc/VectorizedOrcAcidRowBatchReader.java

```diff
@@ -680,14 +681,15 @@ public void setBaseAndInnerReader(
    * @param path The Orc file path we want to get the OrcTail for
    * @param conf The Configuration to access LLAP
    * @param cacheTag The cacheTag needed to get OrcTail from LLAP IO cache
+   * @param fileKey fileId of the Orc file (either the Long fileId of HDFS or the SyntheticFileId)
    * @return ReaderData object where the orcTail is not null. Reader can be null, but if we had to create
    * one we return that as well for further reuse.
    */
-  private static ReaderData getOrcTail(Path path, Configuration conf, CacheTag cacheTag) throws IOException {
+  private static ReaderData getOrcTail(Path path, Configuration conf, CacheTag cacheTag, Object fileKey) throws IOException {
```

Comment:
Maybe 2 different getOrcTail method on the LLAP IO interface? @szlta?

Reply:
I am not sure, we will ever want to use it. without fileId.

Reply:
I'm fine with leaving just this one endpoint, and fileKey being optionally null.
Please do emphasize the optionality of fileKey arg in the javadoc part: aka it will be generated if not given, etc..